### PR TITLE
MC-37718: Grouped product remains In Stock On Mass Update

### PR DESCRIPTION
**Pull request by nikita-shcherbatykh**
### Related Pull Requests
- https://github.com/magento-tsg-csl3/magento2ce/pull/525

### Description (*)

Steps to reproduce:

Create two simple products with stock.
Create Grouped product and assign two previously created simple products.
select 2 simple prod, Actions> click on Update attribute.
click on tab2 advanced inventory.uncheck the box for qty "0" and save
Run bin/magento cron:run
Actual Res: Stock status = "In stock" 

Expected: Stock Status = "Out of stock"

### Fixed Issues (if relevant)

1. Fixes magento/inventory#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by nikita-shcherbatykh** [comment]
@magento run all tests

**Comment by nikita-shcherbatykh** [comment]
@magento run all tests

**Comment by nikita-shcherbatykh** [comment]
@magento run all tests

File: InventoryCatalog/Model/UpdateInventory.php [hunk 1]
```diff
@@ -119,9 +122,9 @@ public function __construct(
      * Update legacy stock items, default source items and reindex inventory for given product skus.
      *
      * @param InventoryData $data
-     * @return void
+     * @return array
      */
-    public function execute(InventoryData $data): void
+    public function execute(InventoryData $data)
```
Comment (sidolov) [inline InventoryCatalog/Model/UpdateInventory.php]: Such changes contradict with CQRS principles, commands shouldn't return any value.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**